File: DB.Infra/Context/Campaign.cs
using System;
using System.Collections.Generic;

namespace DB.Infra.Context;

public partial class Campaign
{
    public long CampaignId { get; set; }

    public long UserId { get; set; }

    public string Title { get; set; }

    public string Description { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public int Status { get; set; }

    public bool NameRequired { get; set; }

    public bool EmailRequired { get; set; }

    public bool PhoneRequired { get; set; }

    public int? MinAge { get; set; }

    public string BgImage { get; set; }

    public string TopImage { get; set; }

    public string YoutubeUrl { get; set; }

    public string CustomCss { get; set; }

    public int? MinEntry { get; set; }

    public virtual ICollection<CampaignEntry> CampaignEntries { get; set; } = new List<CampaignEntry>();

    public virtual ICollection<CampaignField> CampaignFields { get; set; } = new List<CampaignField>();

    public virtual ICollection<Client> Clients { get; set; } = new List<Client>();

    public virtual User User { get; set; }
}

File: viralt-app/src/types/campaign.ts
/** Campaign Types — Types for the campaign system */

/** Main campaign information */
export interface CampaignInfo {
  campaignId: number;
  userId: number;
  title: string;
  description: string;
  startTime: string | null;
  endTime: string | null;
  status: number;
  nameRequired: boolean;
  emailRequired: boolean;
  phoneRequired: boolean;
  minAge: number | null;
  bgImage: string;
  topImage: string;
  youtubeUrl: string;
  customCss: string;
  minEntry: number | null;
}

/** Data required to create a new campaign (no ID) */
export interface CampaignInsertInfo {
  userId: number;
  title: string;
  description: string;
  startTime: string | null;
  endTime: string | null;
  status: number;
  nameRequired: boolean;
  emailRequired: boolean;
  phoneRequired: boolean;
  minAge: number | null;
  bgImage: string;
  topImage: string;
  youtubeUrl: string;
  customCss: string;
  minEntry: number | null;
}

/** Data required to update an existing campaign (includes ID) */
export interface CampaignUpdateInfo extends CampaignInsertInfo {
  campaignId: number;
}

/** Campaign list API result */
export interface CampaignListResult {
  campaigns: CampaignInfo[];
  sucesso: boolean;
  mensagem: string | null;
  erros: string[] | null;
}

/** Campaign get API result */
export interface CampaignGetResult {
  campaign: CampaignInfo | null;
  sucesso: boolean;
  mensagem: string | null;
  erros: string[] | null;
}

/** Status-only result (reuse from user types if preferred) */
export interface StatusResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
}
